src/gwt/src/org/rstudio/core/client/virtualscroller/virtualscroller.js

```diff
@@ -87,7 +87,12 @@ var VirtualScroller;
       }
 
       // jump to latest button
-      if (this.scrollerEle.parentElement.getElementsByClassName("jump-to-latest-console").length < 1) {
+      var hasJumpToConsoleBtn =
+        this.scrollerEle &&
+        this.scrollerEle.parentElement &&
+        this.scrollerEle.parentElement.getElementsByClassName("jump-to-latest-console").length < 1;
```

Comment:
Unfortunately these invariants will just hide bugs. Those two cases are invalid for how this is written. A scroller element with a parent is necessary for the virtualizer, at least at the moment. We could put a check higher up to throw an error early, like when it's traversing up the DOM to find the scroller element, but the assumption is that this class is written expressly to work inside an `ace_scroller` element.

Reply:
Roger that. I've made a change so that we throw a plain error if we cannot resolve the scroller element, so at least errors will be more "obvious" for us. Let me know if this looks okay.